CatNote.API/Mappers/MapperApiProfile.cs

```diff
@@ -10,6 +13,49 @@ public MapperApiProfile()
     {
         CreateMap<UserDTO, UserModel>().ReverseMap();
         CreateMap<TaskDTO, TaskModel>().ReverseMap();
-        CreateMap<AchievementDTO, AchievementModel>().ReverseMap();
+        CreateMap<AchievementDTO, Achievement>().ConvertUsing<AchievementResolver>();
+        CreateMap<Achievement, AchievementDTO>().ConvertUsing<AchievementEntityResolver>();
+    }
+}
+
+public class AchievementEntityResolver : ITypeConverter<Achievement, AchievementDTO>
+{
+    public AchievementDTO Convert(Achievement source, AchievementDTO destination, ResolutionContext context)
+    {
+        return new AchievementDTO
+        {
```

Comment:
Why do you need resolver here? This model is not so complex. You could just use

.ForMember(d => d.Id, o => o.MapFrom(s => s.AchievementId));